src/absil/illib.fs

```diff
@@ -20,9 +25,10 @@ let (>>>&) (x:int32) (n:int32) = int32 (uint32 x >>> n)
 let notlazy v = Lazy<_>.CreateFromValue v
 
 let inline isNonNull x = not (isNull x)
-let inline nonNull msg x = if isNull x then failwith ("null: " ^ msg) else x
+let inline nonNull msg x = if isNull x then failwith ("null: " + msg) else x
 let (===) x y = LanguagePrimitives.PhysicalEquality x y
 
+#if !FABLE_COMPILER
```

Comment:
(To give context: we've had this before, e.g. for Silverlight we scattered hundreds of ``#if SILVERLIGHT`` across the codebase.  It took years and years to divide them between the various categories - ones that carried across to .NET Core, and ones that were feature-specific etc.   The lesson I learned from that is to always ask people to be specific about why `#if` creep in: if there are just 5-10 uses of an `#if`` it's ok, but there are about 180 uses of ``if FABLE_COMPILER`` - so we should take the time to divide them into the true underlying reasons now)